src/reconcile/rec_child.c

```diff
@@ -128,6 +134,12 @@ __rec_child_deleted(
         cmsp->del = *page_del;
         cmsp->state = WT_CHILD_PROXY;
         page_del->selected_for_write = true;
+        /*
+         * We rely on the next reconciliation to remove the deleted entries if they become globally
+         * visible. Mark the page dirty to ensure we will trigger the next reconciliation on this
+         * page. Otherwise, the deleted pages may never be removed from disk.
+         */
+        r->leave_dirty = true;
```

Comment:
Won't this change mean that an internal page can never be evicted while it has live deleted child pages? That seems too extreme to me

Reply:
```
    /*
     * Internal pages with deletes that aren't stable cannot be evicted, we don't have sufficient
     * information to restore the page's information if subsequently read (we wouldn't know which
     * transactions should see the original page and which should see the deleted page).
     */
    if (F_ISSET(r, WT_REC_EVICT))
        return (__wt_set_return(session, EBUSY));

    /* If the page cannot be marked clean. */
    r->leave_dirty = true;
```
In 6.0, we are not allow to evict any internal pages with deleted pages and checkpoint will always mark the page dirty. I think now we should allow to evict but checkpoint should still mark the page dirty?

Reply:
Marking the page dirty in reconciliation means that it won't be evicted.
I think (though I don't remember exactly) that some pages couldn't be evicted before, but they can now).
It might not be ok for MongoDB (not standalone), since it still won't include version information on the page.

Reply:
OK, if checkpoint writes it, mongodb still loses the timestamp information. So mongodb can still only truncate without a timestamp, is that correct?